frontend/public/components/monitoring/utils.ts

```diff
@@ -63,7 +63,7 @@ export const getAlertsAndRules = (
   });
 
   // Add `rule` object to each alert
-  const alerts = _.flatMap(rules, (rule) => rule.alerts.map((a) => ({ rule, ...a })));
+  const alerts = _.flatMap(rules, (rule) => rule.alerts?.map((a) => ({ rule, ...a })));
```

Comment:
Now, we are getting empty array instead of null if alerts is not present for the rule.

```suggestion
  const alerts = _.flatMap(rules, (rule) => rule.alerts.map((a) => ({ rule, ...a })));
```